### PR TITLE
Adds BackupRetentionPeriod to RDS spec

### PR DESCRIPTION
**Pull request by willdavsmith**
* Adds BackupRetentionPeriod to AWS RDS recipe

### CODE REVIEW

File: aws/sqldatabases.bicep [hunk 1]
```diff
@@ -50,6 +50,9 @@ param allocatedStorage string = '20'
 @description('Database license model')
 param licenseModel string = 'license-included'
 
+@description('The number of days for which automated backups are retained. Setting this parameter to a positive number enables backups. Defaults to 0.')
+param backupRetentionPeriod int = 0
```
Comment (sylvainsf) [inline aws/sqldatabases.bicep:54]: The default of 0 means indefinite.

Comment (kachawla) [inline aws/sqldatabases.bicep:54]: Can we set the default to 1 (same as what RDS sets https://docs.aws.amazon.com/cli/latest/reference/rds/create-db-instance.html), and override it to zero for our eshop scenario?

Comment (willdavsmith) [reply aws/sqldatabases.bicep:54]: AWS CloudFormation spec says 0 should mean disabled backups: https://us-west-2.console.aws.amazon.com/cloudformation/home?region=us-west-2#/registry/activated-extensions/details/schema?arn=arn%3Aaws%3Acloudformation%3Aus-west-2%3A%3Atype%2Fresource%2FAWS-RDS-DBInstance

Comment (ytimocin) [reply aws/sqldatabases.bicep:54]: Should we also add the upper and lower limits to the description?

https://docs.aws.amazon.com/sdkfornet1/latest/apidocs/html/P_Amazon_RDS_Model_ModifyDBInstanceRequest_BackupRetentionPeriod.htm

**Review by kachawla — COMMENTED** [review]
@willdavsmith is there something blocking this PR?

File: aws/sqldatabases.bicep [hunk 2]
```diff
@@ -50,6 +50,9 @@ param allocatedStorage string = '20'
 @description('Database license model')
 param licenseModel string = 'license-included'
 
+@description('The number of days for which automated backups are retained. Setting this parameter to a positive number enables backups. Defaults to 1.')
```
Comment (kachawla) [inline aws/sqldatabases.bicep:53]: Could you include this in the suggestion as well https://github.com/radius-project/recipes/pull/50#discussion_r1399735395?

**Review by kachawla — APPROVED** [review]

File: aws/sqldatabases.bicep (hunk 2 above)
Comment (willdavsmith) [reply aws/sqldatabases.bicep:53]: I checked the spec for this resource and it doesn't look like there's a maximum - I think the link yetkin linked is the C# SDK

```json
    "BackupRetentionPeriod": {
      "type": "integer",
      "minimum": 0,
      "default": 1,
      "description": "The number of days for which automated backups are retained. Setting this parameter to a positive number enables backups. Setting this parameter to 0 disables automated backups."
    },
```

https://us-west-2.console.aws.amazon.com/cloudformation/home?region=us-west-2#/registry/activated-extensions/details/schema?arn=arn%3Aaws%3Acloudformation%3Aus-west-2%3A%3Atype%2Fresource%2FAWS-RDS-DBInstance

Comment (kachawla) [reply aws/sqldatabases.bicep:53]: https://docs.aws.amazon.com/cli/latest/reference/rds/create-db-instance.html - this describes max as 35. 

```
Constraints:

Must be a value from 0 to 35.
Can't be set to 0 if the DB instance is a source to read replicas.
Can't be set to 0 for an RDS Custom for Oracle DB instance.
```

**Pull request merged**